File: solutions/elixir/lasagna/2/lib/lasagna.ex
defmodule Lasagna do
  def expected_minutes_in_oven, do: 40

  def remaining_minutes_in_oven(curr_mins), do: expected_minutes_in_oven() - curr_mins
  
  def preparation_time_in_minutes(layers), do: 2 * layers
  
  def total_time_in_minutes(layers, curr_mins), do: preparation_time_in_minutes(layers) + curr_mins
  
  def alarm(), do: "Ding!"
end

File: solutions/java/lasagna/1/src/main/java/Lasagna.java
public class Lasagna {
    // TODO: define the 'expectedMinutesInOven()' method
    public int expectedMinutesInOven() {
        return 40;
    }
    
    // TODO: define the 'remainingMinutesInOven()' method
    public int remainingMinutesInOven(int actualMinutes) {
        return expectedMinutesInOven() -  actualMinutes;
    }
    
    // TODO: define the 'preparationTimeInMinutes()' method
    public int preparationTimeInMinutes(int layersNumber) {
        return layersNumber * 2;
    }
    
    // TODO: define the 'totalTimeInMinutes()' method
    public int totalTimeInMinutes(int layersNumber, int actualMinutes) {
        return preparationTimeInMinutes(layersNumber) + actualMinutes;
    }
}
